lib/data/integrity_checker.rb

```diff
@@ -0,0 +1,30 @@
+require 'auth/authentication_backend'
+# The purpose of this is to make sure any existing Cognito groups
+# have a corresponding local team in DB. If not, it creates them.
+
+class IntegrityChecker
+  include AuthenticationBackend
+
+  def initialize
+    return if SelfService.service(:cognito_stub)
+
+    Rails.logger.info('Checking the data integrity...')
+    check_groups_vs_teams
+  end
+
+  def check_groups_vs_teams
+    cognito_groups = list_groups
+    return if cognito_groups.empty?
+
+    cognito_groups.groups.each do |group|
+      Rails.logger.info("Checking if #{group.group_name} group has a corresponding team in DB...")
+      if Team.exists?(team_alias: group.group_name)
+        Rails.logger.info("OK - #{group.group_name} group has a corresponding team in DB...")
+      else
+        Rails.logger.warn("#{group.group_name} group does not have a corresponding team in DB! Creating...")
+        name = group.description.present? ? group.description : group.group_name
+        NewTeamEvent.create(name: name)
```

Comment:
It's unlikely description will be empty, but if it is and the team is created in our db with the `group_name` I guess we'd need to cater for how it's presented back to the user as it won't have spaces in it etc?